src/api/firebase.js

```diff
@@ -103,12 +104,19 @@ export async function addItem(
 	}
 }
 
-export async function updateItem() {
+export async function updateItem(listToken, itemId, totalPurchases) {
 	/**
 	 * TODO: Fill this out so that it uses the correct Firestore function
 	 * to update an existing item. You'll need to figure out what arguments
 	 * this function must accept!
 	 */
+	// get reference to the item's document in Firestore
+	const itemDocRef = doc(db, listToken, itemId);
+	// set dateLastPurchased to current date, incrememt totalPurchases
+	await updateDoc(itemDocRef, {
+		dateLastPurchased: new Date(),
+		totalPurchases: totalPurchases + 1,
```

Comment:
Nice work setting this all up! The only thing I wanted to mention is that if a user accidentally marks an item as purchased, with this current logic, they cannot undo it. We can probably add that to a separate issue since it's not part of this week's acceptance criteria.

Reply:
Thanks! Yeah @Amanda2900 and I thought about whether we wanted to do it as part of this issue after discussing the undo functionality at our kickoff on Sunday. Due to how much it would change the shape of the data and affect other components / functions, we decided it might be best to stick to the requirements for this week, and possibly add that later. I would love to see that functionality get added from a user experience perspective though, and would be happy to work on it if we add it as another later issue.